packages/buttons/src/styled/StyledIconButton.ts

```diff
@@ -69,17 +69,17 @@ const iconStyles = (props: IButtonProps & ThemeProps<DefaultTheme>) => {
   `;
 };
 
-export const StyledIconButton = styled(StyledButton as 'button').attrs({
-  'data-garden-id': COMPONENT_ID,
+export const StyledIconButton = styled(StyledButton).attrs<IButtonProps>(props => ({
+  'data-garden-id': props['data-garden-id'] || COMPONENT_ID,
```

Comment:
As mentioned in this [comment](https://github.com/zendeskgarden/react-components/pull/1807/files/3464347fc3856858345009856e5bde2901e032f5#r1602045804), this pattern causes a regression with `IconButton`. `StyledIconButton` now inherits the `COMPONENT_ID` `buttons.button` from `StyledButton`. It should be: 

https://github.com/zendeskgarden/react-components/blob/52822988c09ec0bbcf0a82f63d3d8622b74d0b98/packages/buttons/src/styled/StyledIconButton.ts#L14

![Screen Shot 2024-05-15 at 7 37 03 AM](https://github.com/zendeskgarden/react-components/assets/6879688/ef137484-1130-466c-9bf1-82aaa911dd3e)


For `styled-components` extending another `styled-component`, we must first check the `data-garden-id` doesn't match the one set on the `styled-component` it extends.

```suggestion
export const StyledIconButton = styled(StyledButton).attrs<IButtonProps>(props => ({
  'data-garden-id':
    props['data-garden-id'] && props['data-garden-id'] !== 'buttons.button'
      ? props['data-garden-id']
      : COMPONENT_ID,
```

This [sandbox](https://codesandbox.io/p/sandbox/overriding-attrs-2-l3y6mj?file=%2Fsrc%2FExample.tsx%3A131%2C17) highlights all the possible combinations, what works and what doesn't. There's a lot there to unfold. You've been warned. 😅